lib/logitech_receiver/settings_templates.py

```diff
@@ -176,6 +179,42 @@ def _feature_smooth_scroll():
 					label=_SMOOTH_SCROLL[1], description=_SMOOTH_SCROLL[2],
 					device_kind=_DK.mouse)
 
+def _feature_smart_shift():
+	_MIN_SMART_SHIFT_VALUE = 0
+	_MAX_SMART_SHIFT_VALUE = 50
+	class _SmartShiftRW(_FeatureRW):
+		def __init__(self, feature):
+			super(_SmartShiftRW, self).__init__(feature)
+
+		def read(self, device):
+			value = super(_SmartShiftRW, self).read(device)
+			if _bytes2int(value[0:1]) == 1:
+				# Mode = Freespin, map to minimum
+				return _int2bytes(_MIN_SMART_SHIFT_VALUE, count=1)
+			else:
+				# Mode = smart shift, map to the value, capped at maximum
+				threshold = min(_bytes2int(value[1:2]), _MAX_SMART_SHIFT_VALUE)
+				return _int2bytes(threshold, count=1)
+
+		def write(self, device, data_bytes):
+			threshold = _bytes2int(data_bytes)
+			# Freespin at minimum
+			mode = 1 if threshold == _MIN_SMART_SHIFT_VALUE else 2
+
+			# Ratchet at maximum
+			if threshold == _MAX_SMART_SHIFT_VALUE:
+				threshold = 255
+
+			data = _int2bytes(mode, count=1) + _int2bytes(threshold, count=1) * 2
```

Comment:
(This sets both the current and (afer reset/default) setting for when the ratchet automatically disengages.)